custom-schemes.md

```diff
@@ -8,19 +8,25 @@ Existing specifications like [OpenID4VP](https://openid.net/specs/openid-4-verif
    - This seems to increase the opportunity for attacks such as replay attacks, man-in-the-middle attacks and phishing. It also seems to limit the opportunity for abuse detection and prevention.
    - A browser API can securely pass the requesting origin on to the wallet so that the wallet can know with confidence what origin the user was on when the request was initiated.
    
-## How are wallet apps supposed to pass info back to the invoking page?
-
-  - Does the page give the wallet a URL to navigate back to as part of the custom scheme URL it uses to launch the wallet?
-
-  - If so, that's a problem because it won't go back to the same tab. Apps don't have the ability to return information to the same page that invoked them after a custom scheme invocation; they can just open a new page in a new tab.
-
 ## Can wallets limit requests to secure contexts?
 
    - If a user uses a website via http (not https) and it invokes a custom scheme, it's possible that an attacker with access to the network could eavesdrop on or even modify the request (such as by replacing the verifier's public key with their own).
    - A browser API could (and almost certainly should) take the position that it will function only within the context of a secure context (https).
 
 ## User experience concerns
 
+### What is the user experience of the wallet operation?
+
+   - Launching an entirely new application can be jarring for users, taking them out of the flow of the journey they were trying to complete on a website. This risks abandonment, and so is a disincentive for verifiers to invoke the flow.
+   - A browser API could integrate with the operating system to provide a less jarring wallet overlay on top of the web page, keeping the user feeling in flow with the site's experience.
```

Comment:
Yes! Exactly.

What we might be looking for here is something like the "peek" feature in the Arc browser:

https://www.reddit.com/r/ArcBrowser/comments/113ui8u/peek_by_arc_the_best_tab_never_made/

In fact, that would allow a web-based wallet to pop up in first party mode /and/ a native app to pop up using the same UI (a good thing). The only thing missing would be the "cross-tab" communication between the initiating website and the initiated digital wallet.

If we're looking for the "minimal thing that could be done to enable the ecosystem", this might be it. To be clear, you would still do "wallet/credential selection" in this "peek overlay" (to hide the interaction from all wallets until the user makes a choice)... but then, you don't do a jarring full-screen switch -- you just show the wallet content in the "peek overlay" and once the interaction is done, you get rid of the overlay and give the website the result (resolve the Promise w/ the data from the wallet).

Reply:
Yep. FWIW I've had a "peak overlay" UI experiment [on my back-burner](https://bugs.chromium.org/p/chromium/issues/detail?id=1364321) for awhile. It seemed to me like it might be possible to make "window.open(...,"popup") just use this by default on mobile. It turned out to be harder to implement than I thought (lots of web platform APIs aren't hooked up in this preview UI we already have in Chrome), but I think this is something Chrome could prioritize if there was sufficient demand.